salt/returners/zabbix_return.py

```diff
@@ -23,34 +23,40 @@
 __virtualname__ = "zabbix"
 
 
+ZABBIX_SENDER_CANDIDATES = [
+    "/usr/local/zabbix/bin/zabbix_sender",
+    "/usr/bin/zabbix_sender",
+]
+
+ZABBIX_CONFIG_CANDIDATES = [
+    "/usr/local/zabbix/etc/zabbix_agentd.conf",
+    "/usr/local/zabbix/etc/zabbix_agent2.conf",
+    "/etc/zabbix/zabbix_agentd.conf",
+    "/etc/zabbix/zabbix_agent2.conf",
+]
+
+ZABBIX_SENDER = next((p for p in ZABBIX_SENDER_CANDIDATES if os.path.exists(p)), None)
+ZABBIX_CONFIG = next((p for p in ZABBIX_CONFIG_CANDIDATES if os.path.exists(p)), None)
+
+
 def __virtual__():
-    if zbx():
-        return True
-    return False, "Zabbix returner: No zabbix_sender and zabbix_agend.conf found."
-
-
-def zbx():
-    if os.path.exists("/usr/local/zabbix/bin/zabbix_sender") and os.path.exists(
-        "/usr/local/zabbix/etc/zabbix_agentd.conf"
-    ):
-        zabbix_sender = "/usr/local/zabbix/bin/zabbix_sender"
-        zabbix_config = "/usr/local/zabbix/etc/zabbix_agentd.conf"
-        return {"sender": zabbix_sender, "config": zabbix_config}
-    elif os.path.exists("/usr/bin/zabbix_sender") and os.path.exists(
-        "/etc/zabbix/zabbix_agentd.conf"
-    ):
-        zabbix_sender = "/usr/bin/zabbix_sender"
-        zabbix_config = "/etc/zabbix/zabbix_agentd.conf"
-        return {"sender": zabbix_sender, "config": zabbix_config}
-    else:
-        return False
+    if ZABBIX_SENDER is None or ZABBIX_CONFIG is None:
+        message = "No zabbix_sender and no zabbix_agent(d|2).conf found."
```

Comment:
Would be clearer as `if return elif return else return`.

Consider also glob syntax for the message rather than regex syntax - i.e. `{d,2}`

Reply:
Thanks for the feedback. I prefer the current approach. Is this a non-mandatory suggestion or a must? :)

Reply:
Well I got confused reading it.
None of my suggestions are mandatory, but I strongly feel that a simpler control flow is better.